types/Tracker/TrackingDetail.d.ts

```diff
@@ -0,0 +1,33 @@
+import { IBaseObject } from '../base';
+import { TTrackerStatus } from './TrackerStatus';
+import { ITrackingLocation } from './TrackingLocation';
+
+/**
+ * @see https://www.easypost.com/docs/api/node#tracking-detail-object
+ */
+export declare interface ITrackingDetail extends IBaseObject<'TrackingDetail'> {
+  /**
+   * Description of the scan event
+   */
+  message: string;
+
+  /**
+   * Status of the package at the time of the scan event, possible values are "unknown", "pre_transit", "in_transit", "out_for_delivery", "delivered", "available_for_pickup", "return_to_sender", "failure", "cancelled" or "error"
+   */
+  status: TTrackerStatus;
+
+  /**
+   * The timestamp when the tracking scan occurred
+   */
+  datetime: string; // TODO: maybe type should be Date?
```

Comment:
That depends. Our API docs state this field as a `datetime` field; however, our API most likely losely translates this on the backend and would accept a string of a date. Does Typescript have a native datetime type?

Reply:
Well if it just accepts an ISO string like `"datetime": "2022-06-29T00:00:00.000"` then it would just be a `string`, but if this would accept a javascript `new Date()`, then it would be `datetime: Date`

Reply:
All my usage of the API would lead me to string. I've removed the TODO.